src/transformers/parsing-utilities.ts

```diff
@@ -100,18 +105,26 @@ export function NamedDeclaration(
 
 export function DefaultDeclaration(
   context: PluginContext,
+  id: string,
   declaration: ExportDefaultDeclaration,
 ): ExportNameToClosureMapping | null {
   if (declaration.declaration) {
     switch (declaration.declaration.type) {
       case 'FunctionDeclaration':
-        const functionName = functionDeclarationName(context, declaration);
+        const functionName = functionDeclarationName(context, id, declaration);
         if (functionName !== null) {
           return {
             [functionName]: ExportClosureMapping.NAMED_DEFAULT_FUNCTION,
           };
+        } else {
+          // When Rollup encounters a default export that is unnamed,
+          // it uses camelCase(filename) of the id to name the function.
+          // THIS IS NOT INTUITIVE!
```

Comment:
Since this logic is not intuitive, and pretty rollup specific... lets move it to a specific method with a test.